src/sdk/main/src/impl/HttpClient.cc

```diff
@@ -18,101 +18,77 @@
  *
  */
 
-#include "exceptions/CURLException.h"
-
 #include "impl/HttpClient.h"
 
+#include <httplib.h>
+#include <stdexcept>
+#include <string>
+#include <string_view>
+
 namespace Hedera::internal
 {
-HttpClient::HttpClient()
+namespace
 {
-  if (curl_global_init(CURL_GLOBAL_DEFAULT) != 0)
-  {
-    throw CURLException("Failed to initialize libcurl!");
-  }
-}
-
-HttpClient::~HttpClient()
+// The index in a URL to begin searching for the path after the end of the URL scheme ("http://" or "https://").
+const int SCHEME_END_INDEX = 8;
+
+//
+// Perform an HTTP request.
+//
+// @param url    The URL to which to send the request.
+// @param method The HTTP method type of this request.
+// @param body   The body of the request.
+// @return The response of the request.
+//
+[[nodiscard]] std::string performRequest(std::string_view url, std::string_view method, std::string_view body)
 {
-  curl_global_cleanup();
-}
+  // Create an HTTP client to communicate with the given URL.
+  httplib::Client client(std::string(url.substr(0, url.find('/', SCHEME_END_INDEX))));
+  const std::string path = url.substr(url.find('/', SCHEME_END_INDEX)).data();
 
-// example mirrorNode query:
-// httpClient.invokeREST("https://testnet.mirrornode.hedera.com/api/v1/accounts/" + newAccountId ,"GET", "");
-// note: should time out before calling this function because the mirror node is not updated on time if accountID has
-// been created exactly before the call. Works without timeout if the data in the mirror node is there from some seconds
-// beforehand
-std::string HttpClient::invokeREST(const std::string& url,
-                                   const std::string& httpMethod,
-                                   const std::string& requestBody)
-{
-  std::unique_ptr<CURL, decltype(&curl_easy_cleanup)> curl(curl_easy_init(), curl_easy_cleanup);
+  httplib::Result res;
 
-  if (!curl)
+  // Perform the request based on the HTTP method
```

Comment:
nit: Maybe would look better if we have:
```
const std::unordered_set<std::string_view> supportedMethods = { "GET", "POST" };
```
Check against the set and throw if not in the set.